src/pp.js

```diff
@@ -211,7 +221,10 @@ function fits(next, restCommands, width) {
   return false;
 }
 
-function print(w, doc) {
+function print(options, doc) {
```

Comment:
The crux of the issue is that we are doing repeatIndent from 0 for every single line that we are printing.

What I would recommend you to do is to make an indent cache inside of this function that looks like this:


```js
let indentCache = [];
function _makeIndent(n) {
  let cached = indentCache[n];
  if (cached) { return cached; }

  let result = '';
  for (let i = 0; i < n; ++i) {
    result += " ";
  }
  indentCache[n] = result;
  return result;
}
```

(of course tweak this again when you need to change the spacing to be tabs).

What do you think?